core/src/client/lwm2m_static.c

```diff
@@ -599,4 +599,53 @@ AwaError AwaStaticClient_RegisterResourceWithHandler(AwaStaticClient * client, c
                                             NULL, 0 , 0);
 }
 
+AwaError AwaStaticClient_ResourceChanged(AwaStaticClient * client, AwaObjectID objectID, AwaObjectInstanceID objectInstanceID, AwaResourceID resourceID)
+{
+    AwaError result = AwaError_Unspecified;
+
+    if (client != NULL)
+    {
+        ResourceDefinition * definition = Definition_LookupResourceDefinition(Lwm2mCore_GetDefinitions(client->Context), objectID, resourceID);
+        if (definition != NULL)
+        {
+            if (definition->Handlers.Read == NULL)
+            {
+                if (definition->Handler != NULL)
+                {
+                    void * valueBuffer = NULL;
+                    int valueBufferSize = 0;
+                    int resourceInstanceID = 0;  // FIXME
+
+                    if (definition->Handler(client, AwaOperation_Read, objectID, objectInstanceID, resourceID, resourceInstanceID, (void **)&valueBuffer, &valueBufferSize, NULL) == AwaLwm2mResult_SuccessContent)
+                    {
+                        Lwm2m_MarkObserversChanged(client->Context, objectID, objectInstanceID, resourceID, valueBuffer, valueBufferSize);
+                        result = AwaError_Success;
+                    }
+                    else
+                    {
+                        result = AwaError_Internal;
```

Comment:
Could be useful to print the resulting error code from the Read before throwing it away (debug level)